deploy_tools/deploy.py

```diff
@@ -1,14 +1,69 @@
+from typing import Dict
+from web3.contract import Contract
 from web3 import Web3
+from web3.utils.transactions import fill_nonce
 
 
-def deploy_compiled_contract(*, abi, bytecode, web3, constructor_args=()):
+def deploy_compiled_contract(
+    *,
+    abi,
+    bytecode,
+    web3: Web3,
+    constructor_args=(),
+    transaction_options: Dict = None,
+    private_key=None,
+) -> Contract:
+    """
+    Deploys a compiled contract either using an account of the node, or a local private key
+
+    Returns: The deployed contract as a web3 contract
+
+    """
     contract = web3.eth.contract(abi=abi, bytecode=bytecode)
-    txhash = contract.constructor(*constructor_args).transact()
-    receipt = wait_for_successful_transaction_receipt(web3, txhash)
+    constuctor_call = contract.constructor(*constructor_args)
+
+    receipt = send_contract_call(
+        constuctor_call,
+        web3=web3,
+        transaction_options=transaction_options,
+        private_key=private_key,
+    )
+
     address = receipt["contractAddress"]
     return contract(address)
 
 
+def send_contract_call(
```

Comment:
I would rename this to `send_contract_transaction`. I would expect a call to have no impact on the chain, and this sends a transaction to the chain.

Reply:
call are used in several ways. One is a function call, which can change the chain of course, and that is what I mean here. But I see can see the confusion. My problem is that I dont know how to proberly name this object coming from web3 (`contract.functions.foo()`). It is not really a transaction, thats why I hesitate to call it `send_contract_transaction`.

Reply:
will change it to `send_function_call_transaction`